zerver/tests/test_markdown.py

```diff
@@ -3099,6 +3099,28 @@ def test_topic_multiple(self) -> None:
             ".</p>",
         )
 
+    def test_message_id_multiple(self) -> None:
+        denmark = get_stream("Denmark", get_realm("zulip"))
+        sender_user_profile = self.example_user("othello")
+        msg = Message(
+            sender=sender_user_profile,
+            sending_client=get_client("test"),
+            realm=sender_user_profile.realm,
+        )
+        content = "As mentioned in #**Denmark>danish@123** and #**Denmark>danish@456**."
+        self.assertEqual(
+            render_message_markdown(msg, content).rendered_content,
+            "<p>As mentioned in "
+            f'<a class="stream-topic-message" data-stream-id="{denmark.id}" '
+            f'href="/#narrow/stream/{denmark.id}-{denmark.name}/topic/danish/near/123">'
+            f"#Denmark &gt; danish @ 123</a>"
+            " and "
+            f'<a class="stream-topic-message" data-stream-id="{denmark.id}" '
+            f'href="/#narrow/stream/{denmark.id}-{denmark.name}/topic/danish/near/456">'
+            f"#Denmark &gt; danish @ 456</a>"
+            ".</p>",
```

Comment:
This is probably a question for the design thread, but I'm not sure we  want the message ID in the human-visible text?

In any case, the other thing we should make sure to test is what happens with an `@` in the middle of the topic; presumably we should be able to handle it reasonable as long as it't not followed by something like `3**`, which would be pretty weird.

Reply:
As per the suggestion on CZO, we can do away with these backend changes entirely, and just transform /near links the same way we did in #29302. I have updated the PR with that.

Reply:
So I think we definitely want the Backend markdown processor approach; the big benefit of that design is that you get much more visually appealing content in the compose box before sending the message.

Reply:
Okay, so should we make it so that we can paste a link to a message and it renders as `#design>abc@💬`, but the actual message text remains the same (ie, the link)? Or do we transform the pasted URL into `#design>abc@123` on pasting and then replace the message id with `💬` on rendering?

The second approach sounds better.